vagrant/src/main/java/org/jclouds/vagrant/internal/VagrantNodeRegistry.java

```diff
@@ -73,36 +93,41 @@ public long getDelay(TimeUnit unit) {
          return unit.convert(expiryTime - timeSupplier.get(), TimeUnit.MILLISECONDS);
       }
    }
-   private Map<String, VagrantNode> nodes = new ConcurrentHashMap<String, VagrantNode>();
-   private DelayQueue<TerminatedNode> terminatedNodes = new DelayQueue<TerminatedNode>();
+
+   private final DelayQueue<TerminatedNode> terminatedNodes = new DelayQueue<TerminatedNode>();
```

Comment:
Hmmm... AFAIK that only happens in AWS because the nodes don't disappear directly. jclouds expects the nodes to be gone or in terminated state (as per the [node terminated predicate](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/predicates/AtomicNodeTerminated.java)).
This looks great and we've learned something! But we'd better keep the code as simple as possible. Could you try removing the terminated node cache and test the provider if you just completely remove the machines? It should still work as expected.